libnetwork/resolvconf/resolv.go

```diff
@@ -30,17 +30,28 @@ type Params struct {
 	// IPv6Enabled will filter ipv6 nameservers when not set to true.
 	IPv6Enabled bool
 	// KeepHostServers can be set when it is required to still keep the
-	// original resolv.conf content even when custom Nameserver/Searches/Options
+	// original resolv.conf nameservers even when explicit Nameservers
 	// are set. In this case they will be appended to the given values.
 	KeepHostServers bool
+	// KeepHostSearches can be set when it is required to still keep the
+	// original resolv.conf search domains even when explicit search domains
+	// are set in Searches.
+	KeepHostSearches bool
+	// KeepHostOptions can be set when it is required to still keep the
+	// original resolv.conf options even when explicit options are set in
+	// Options.
+	KeepHostOptions bool
 	// Nameservers is a list of nameservers the container should use,
-	// instead of the default ones from the host.
+	// instead of the default ones from the host. Set KeepHostServers
+	// in order to also keep the hosts resolv.conf nameservers.
```

Comment:
Just a nitpick, but it should be "host's". At least I think so - not a native speaker :-)